### PR TITLE
Add float8 example

### PR DESCRIPTION
**Pull request by james77777778**
I've also created a colab for this example:
https://colab.research.google.com/drive/1U0zZ1DqmqJgOHdnmZVk2cNFITONyJXaR

cc @fchollet

### CODE REVIEW

File: examples/keras_recipes/float8_training_and_inference_with_transformer.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+"""
+## Model
+
+Let's build a simple Transformer model. `TransformerBlock` outputs one vector
+for each time step of our input sequence. Here, we take the mean across all time
+steps and use a feedforward network on top of it to classify text.
+"""
+
+
+class TransformerBlock(keras.layers.Layer):
```
Comment (fchollet) [inline examples/keras_recipes/float8_training_and_inference_with_transformer.py:110]: You can just grab this class from KerasNLP -- let's keep the code as short as possible, and focused on the float8 training part.

Comment (james77777778) [reply examples/keras_recipes/float8_training_and_inference_with_transformer.py:110]: Actually, there is an issue in KerasNLP:
https://github.com/keras-team/keras-nlp/pull/1623

TLDR: We need to explicitly expose `training=None` in the signature of `call` in subclasses. This will also fail the float8 training.

File: examples/keras_recipes/float8_training_and_inference_with_transformer.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+Let's load the IMDB dataset from `keras.datasets`. This dataset consists of
+25,000 movie reviews labeled by sentiment (positive/negative). To simplify the
+task, we only consider the top 20k words and the first 200 words of each review.
+"""
+
+vocabulary_size = 20000
+max_sequence_length = 200
+index_from = 3
+(x_train, y_train), (x_val, y_val) = keras.datasets.imdb.load_data(
```
[the remark this answers is not in the document]
Comment (james77777778) [reply examples/keras_recipes/float8_training_and_inference_with_transformer.py]: I've updated that part to be similar to this example:
https://keras.io/examples/nlp/fnet_classification_with_keras_nlp/#loading-the-dataset

File: examples/keras_recipes/float8_training_and_inference_with_transformer.py (hunk 1 above)
Comment (james77777778) [reply examples/keras_recipes/float8_training_and_inference_with_transformer.py]: I've reimplemented the class like this:
```python
class TransformerDecoder(keras_nlp.layers.TransformerDecoder):
    def call(
        self,
        decoder_sequence,
        encoder_sequence=None,
        decoder_padding_mask=None,
        decoder_attention_mask=None,
        encoder_padding_mask=None,
        encoder_attention_mask=None,
        self_attention_cache=None,
        self_attention_cache_update_index=None,
        cross_attention_cache=None,
        cross_attention_cache_update_index=None,
        use_causal_mask=True,
        training=None,
    ):
        return super().call(
            decoder_sequence,
            encoder_sequence,
            decoder_padding_mask,
            decoder_attention_mask,
            encoder_padding_mask,
            encoder_attention_mask,
            self_attention_cache,
            self_attention_cache_update_index,
            cross_attention_cache,
            cross_attention_cache_update_index,
            use_causal_mask,
        )
```

Comment (fchollet) [reply examples/keras_recipes/float8_training_and_inference_with_transformer.py]: I would recommend fixing the issue in KerasNLP first, then relying on plain KerasNLP classes in the example.

Comment (james77777778) [reply examples/keras_recipes/float8_training_and_inference_with_transformer.py]: It should be fixed now. I have updated the example.

**Review by fchollet — APPROVED** [review]
Excellent -- the code and copy is looking great. Please add the generated files.

**Comment by james77777778** [comment]
> Excellent -- the code and copy is looking great. Please add the generated files.

Added

I have reduced the output string by adding `-q` to `wget`
